tdms/tests/unit/array_tests/test_XYZTensor3D.cpp

```diff
@@ -0,0 +1,59 @@
+/**
+ * @file test_XYZTensor3D.cpp
+ * @author William Graham
+ * @brief Tests allocation, deallocation, and methods for the XYZTensor3D class
+ */
+#include <catch2/catch_test_macros.hpp>
+#include <spdlog/spdlog.h>
+#include <complex.h>
+
+#include "arrays.h"
+#include "globals.h"
+
+using namespace std;
+const int REAL = 0, IMAG = 1;
+
+TEST_CASE("Test XYZTensor allocation") {
+  SPDLOG_INFO("== Testing XYZTensor3D allocation/deallocation");
+  // dimensions for the test-tensor
+  int n_layers = 4, n_cols = 8, n_rows = 16;
+
+  // try creating an XYZTensor3D using the default constructor
+  XYZTensor3D<complex<double>> default_constructor_test;
```

Comment:
Marginal comment: worth to _also_ test double? Any value in testing both?
If easy in catch2, I'd say do it. If not: not.